cypress/integration/head_spec.js

```diff
@@ -1,11 +1,14 @@
 describe('Head', function() {
-  it('runs inline scripts', function() {
+  it('runs inline scripts', function(done) {
     cy.visit('http://127.0.0.1:8888/');
 
     cy.contains('Head Link').click();
 
     cy.get('.change-from-inline-script').then(($el) => {
-      expect($el).to.have.text('changed');
+      setTimeout(function(){
+        expect($el).to.have.text('changed');
+        done();
+      }, 100);
```

Comment:
I don't know why this would fix anything, since it already has a wait built in

Reply:
This isn't the critical piece that fixed it. It's this: https://github.com/shanebo/swap/blob/e60e89caa5ab1d6f38e9e6bf5c3ab3157feaeb46/cypress/app/head.js#L1

But even after that, it still had race conditions. This timeout made it never fail.

Reply:
Strange. This should not make any difference at all. All the expects/asserts should wait until they pass, unless they hit the timeout, which is pretty generous.

https://docs.cypress.io/guides/core-concepts/retry-ability.html#Built-in-assertions

I'm not opposed to leaving this in, especially if it solves it, but I would like to get to the bottom of why Cypress' built in feature appears to not be working (such as what conditions cause that). And there are obviously other implications if it doesn't work.

Reply:
I agree, but remember that these are external scripts getting loaded. If there is no JS to run yet, it would just quit the test and fail. Whereas with the timeout it doesn't hit that point until we call done. Look at the other head tests. They function similarly.

Reply:
Based on my understanding of the documentation of how Cypress works, I don't see why those tests need timeouts either.

What are you basing the statement: "If there is no JS to run yet, it would just quit the test and fail." on?

Reply:
Checkout the PR, comment out the setTimeouts and see what happens on your machine with those tests.

Reply:
To be clear, I'm not disputing the efficacy of your solution, I'm just trying to understand why Cypress is not doing what we expect it to be doing

Reply:
I know. I'm saying try it. It wouldn't pass for me without it.

Reply:
OK, I checked it out, and ran them about 20 times with all the timeouts removed.

In that time, they all passed, except for:
- `inserts new css files` failed twice
- `does a hard reload if asset hashes changes` also failed twice, which had no timeout to begin with.

The other two tests never failed for me across about 20 runs.

Reply:
Now run them with timeouts in and see how many times they fail

Reply:
Results:

I ran about 200 tests total between the two
- w/o timeouts: total of 6 failures (including the previous four)
- w/ timeouts, total of 3 failures

Reply:
<img width="863" alt="Screen Shot 2020-12-14 at 1 46 04 PM" src="https://user-images.githubusercontent.com/141022/102127965-c88dc600-3e12-11eb-8ad6-7c470b94b5d0.png">

I wonder if `expect.to.have.text()` is not one of the retry-able assertions/commands?